controller/program/tests/tests/remove_lst.rs

```diff
@@ -0,0 +1,615 @@
+    PoolRebalancing,
+    PoolDisabled,
+}
+
+const MAX_LST_STATES: usize = 10;
+
+fn remove_lst_proptest(
+    pool: PoolState,
+    lsl: LstStateListData,
+    admin: [u8; 32],
+    refund_rent_to: [u8; 32],
+    lst_idx: u32,
+    additional_accounts: impl IntoIterator<Item = PkAccountTup>,
+    error_type: Option<TestErrorType>,
+) -> TestCaseResult {
+    silence_mollusk_logs();
+
+    let LstStateListData { lst_state_list, .. } = lsl;
+
+    let lst_state_list_parsed = LstStatePackedList::of_acc_data(&lst_state_list).unwrap();
+
+    // For invalid index tests, we might not have a valid lst_state
+    let mint = if let Some(lst_state) = lst_state_list_parsed.0.get(lst_idx as usize) {
+        lst_state.into_lst_state().mint
+    } else {
+        // We still need a mint for the instruction keys
+        Pubkey::new_unique().to_bytes()
+    };
+
+    let keys = remove_lst_ix_keys_owned(&admin, &refund_rent_to, &mint, &TOKENKEG_ID);
+
+    let ix = remove_lst_ix(&keys, lst_idx);
+    let mut accounts = remove_lst_fixtures_accounts_opt(&keys);
+
+    // Common upserts
+    upsert_account(
+        &mut accounts,
+        (
+            LST_STATE_LIST_ID.into(),
+            lst_state_list_account(lst_state_list),
+        ),
+    );
+    upsert_account(
+        &mut accounts,
+        (POOL_STATE_ID.into(), pool_state_account(pool)),
+    );
+    upsert_account(
+        &mut accounts,
+        (
+            Pubkey::new_from_array(admin),
+            Account {
+                lamports: u64::MAX,
+                ..Default::default()
+            },
+        ),
+    );
+    upsert_account(
+        &mut accounts,
+        (
+            Pubkey::new_from_array(refund_rent_to),
+            Account {
+                ..Default::default()
+            },
+        ),
+    );
+    upsert_account(
+        &mut accounts,
+        (
+            Pubkey::new_from_array(mint),
+            mock_mint(raw_mint(None, None, u64::MAX, 9)),
+        ),
+    );
+    upsert_account(
+        &mut accounts,
+        (Pubkey::new_from_array(PROTOCOL_FEE_ID), Account::default()),
+    );
+
+    let (pool_reserves_addr, _) = find_pool_reserves_ata(&TOKENKEG_ID, &mint);
+    let (protocol_fee_accumulator_addr, _) = find_protocol_fee_accumulator_ata(&TOKENKEG_ID, &mint);
+
+    upsert_account(
+        &mut accounts,
+        (
+            pool_reserves_addr,
+            mock_token_acc(raw_token_acc(mint, POOL_STATE_ID, 0)),
+        ),
+    );
+    upsert_account(
+        &mut accounts,
+        (
+            protocol_fee_accumulator_addr,
+            mock_token_acc(raw_token_acc(mint, PROTOCOL_FEE_ID, 0)),
+        ),
+    );
+
+    // Additional test-specific upserts
+    additional_accounts
+        .into_iter()
+        .for_each(|account| upsert_account(&mut accounts, account));
+
+    let InstructionResult {
+        program_result,
+        resulting_accounts,
+        ..
+    } = SVM.with(|svm| svm.process_instruction(&ix, &accounts));
```

Comment:
for instructions like this that involves realloc-ing accounts, always important to add the rent exempt check.

Mollusk only handles `InstructionError`s, not `TransactionError`s.

```suggestion
    } = SVM.with(|svm| svm.process_and_validate_instruction(&ix, &accounts, &[Check::all_rent_exempt()]));
```